sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -153,6 +152,27 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
 
       // --- BroadcastHashJoin --------------------------------------------------------------------
 
+      case ExtractEquiJoinKeys(joinType, leftKeys, rightKeys, condition, left, right)
```

Comment:
if users ask to broadcast both the join side in the hint will pick a smaller side to broadcast according to stats